app/serializers/post_stream_serializer_mixin.rb

```diff
@@ -64,11 +64,30 @@ def posts
           serializer.add_raw = true if @options[:include_raw]
           serializer.topic_view = object
 
+          serializer.notice_created_by_users = post_notice_created_by_users if scope.is_staff?
+
           serializer.as_json
         end
       end
   end
 
+  def post_notice_created_by_users
+    @post_notice_created_by_users ||=
+      begin
+        user_ids = []
+        object
+          .post_custom_fields
+          .each do |post_id, custom_fields|
+            if custom_fields[Post::NOTICE]
+              user_ids << custom_fields[Post::NOTICE]["created_by_user_id"]
+            end
+          end
+          .compact
+          .uniq
```

Comment:
I imagine we intended to call `#compact` and `#uniq` on `user_ids`? 🙂 I think we can use [`#filter_map`](https://blog.saeloun.com/2019/05/25/ruby-2-7-enumerable-filter-map/) . Something like this. (Code not tested.)

```ruby
user_ids =
  object
    .post_custom_fields
    .filter_map do |_, custom_fields|
      custom_fields.dig(Post::NOTICE, "created_by_user_id")
    end
    .uniq
```

Reply:
Oh nice ty